asio/CMakeLists.txt

```diff
@@ -0,0 +1,240 @@
+  message(STATUS "disble queue")
+  list(APPEND CPPdefinitions ASIO_DISABLE_KQUEUE)
+endif()
+
+if(ASIO_DISABLE_EPOLL)
+  message(STATUS "disable epoll")
+  list(APPEND CPPdefinitions ASIO_DISABLE_EPOLL)
+endif()
+
+if(ASIO_USE_CLANG_TIDY)
+  message(STATUS "use clang-tidy")
+  find_program(CMAKE_CXX_CLANG_TIDY
+    NAMES clang-tidy clang-tidy-8 clang-tidy-9 clang-tidy-7
+    HINTS /usr/local/bin /opt/local/bin
+  )
+else()
+  unset(CMAKE_CXX_CLANG_TIDY CACHE)
+endif()
+
+if(NOT CONAN_COMPILER)
+  if(NOT ASIO_STANDALONE)
+    #####################################################################
+    # use boost if found
+    option(Boost_USE_MULTITHREADED "Set to OFF to use the non-multithreaded" ON)
+    option(Boost_DEBUG "Set to ON to enable debug output from FindBoost." OFF)
+    option(Boost_DETAILED_FAILURE_MSG "Set to ON to get detailed information" ON)
+    # Set Boost_NO_BOOST_CMAKE to ON to disable the search for boost-cmake.
+    set(Boost_NO_BOOST_CMAKE ON)
+    set(BOOST_ROOT "/opt/local" CACHE PATH "path to the boost lib")
+    set(Boost_USE_STATIC_LIBS OFF)
+    if(WIN32)
+      set(Boost_USE_STATIC_RUNTIME OFF)
+    endif()
+    find_package(Boost 1.67
+        COMPONENTS atomic chrono coroutine context thread system
+    )
+    #####################################################################
+
+    if(Boost_FOUND)
+      include_directories(${Boost_INCLUDE_DIRS})
+      link_directories(${Boost_LIBRARY_DIRS})
+      list(APPEND CPPdefinitions BOOST_ALL_NO_LIB=1
+        BOOST_CHRONO_DONT_PROVIDE_HYBRID_ERROR_HANDLING
+      )
+      list(APPEND EXTRA_LIBS ${Boost_LIBRARIES})
+    endif()
+    #####################################################################
+  endif()
+
+
+  #######################################################################
+  # use openssl if found
+  set(OPENSSL_ROOT_DIR "/usr" CACHE PATH "path to the openssl lib")
+  find_package(OpenSSL HINTS ${OPENSSL_ROOT_DIR} /opt/local)
+  #######################################################################
+
+  if(OpenSSL_FOUND)
+    list(APPEND EXTRA_LIBS ${openSSL_LIBRARIES})
+    message(STATUS "HAVE_OPENSSL")
+    list(APPEND CPPdefinitions HAVE_OPENSSL)
+    include_directories(BEFORE ${openSSL_INCLUDE_DIR})
+  # else()
+  #   find_library(OpenSSL_LIBRARY ssl HINTS ${OpenSSL_ROOT_DIR}/lib)
+  #   find_library(CRYPTO_LIBRARY crypto HINTS ${OpenSSL_ROOT_DIR}/lib)
+  #   if(OpenSSL_LIBRARY AND CRYPTO_LIBRARY)
+  #     set(OpenSSL_FOUND ON)
+  #     message(STATUS "USE CRYPTO_LIBRARY")
+  #     list(APPEND CPPdefinitions HAVE_OPENSSL)
+  #     list(APPEND EXTRA_LIBS ${OpenSSL_LIBRARY} ${CRYPTO_LIBRARY})
+  #     include_directories(BEFORE ${OpenSSL_ROOT_DIR}/include)
+  #   endif()
+  endif()
+  #######################################################################
+else()
+  message(STATUS "ASIO USE CONAN_TARGETS")
+  if(CONAN_LIBS_OPENSSL)
+    list(APPEND CPPdefinitions HAVE_OPENSSL)
+    message(STATUS "have CONAN_LIBS_OPENSSL")
+  endif()
```

Comment:
This is just unnecessary noise. The user already knows what's on his system. Why tell him?